uuid_sd.go

```diff
@@ -22,3 +24,23 @@ func (uuid UUID) shortUUID() (C.ble_uuid_t, uint32) {
 	errCode := C.sd_ble_uuid_vs_add((*C.ble_uuid128_t)(unsafe.Pointer(&uuid[0])), &short._type)
 	return short, errCode
 }
+
+// UUID returns the full length UUID for this short UUID.
+func (s shortUUID) UUID() UUID {
+	var outLen C.uint8_t
+	var outUUID C.uint16_t
+
+	C.sd_ble_uuid_encode(((*C.ble_uuid_t)(unsafe.Pointer(&s))), ((*C.uint8_t)(unsafe.Pointer(&outLen))), ((*C.uint8_t)(unsafe.Pointer(&outUUID))))
```

Comment:
I _think_ you can leave out the `unsafe.Pointer` in most conversions.

```suggestion
	C.sd_ble_uuid_encode(((*C.ble_uuid_t)(&s)), &outLen, ((*C.uint8_t)(unsafe.Pointer(&outUUID))))
```

But unfortunately, this code is unsafe. The `outUUID` (`p_uuid_le`) can be bigger than two bytes for 128-bit UUIDs.

Something like this should be safe:
```suggestion
	if s._type == C.BLE_UUID_TYPE_BLE {
		return New16BitUUID(s.uuid)
	}
	var outUUID UUID
	C.sd_ble_uuid_encode(((*C.ble_uuid_t)(unsafe.Pointer(&s))), ((*C.uint8_t)(unsafe.Pointer(&outLen))), ((*C.uint8_t)(unsafe.Pointer(&outUUID))))
	return outUUID
```

Reply:
Updated, thanks for the "pointers" :joy_cat: